source/agora/consensus/ValidatorSet.d

```diff
@@ -526,6 +526,19 @@ public class ValidatorSet
             return false;
         }
 
+        // check the validity of new pre-image based on the stored pre-image
+        PreimageInfo cur_image;
+        if (this.getPreimage(preimage.enroll_key, cur_image) &&
+            cur_image != PreimageInfo.init)
```

Comment:
Why would the preimage be default-initialized if `getPreimage` returns `true`? It seems to me that should never be the case.

Reply:
> Why would the preimage be default-initialized if getPreimage returns true? It seems to me that should never happen.

It's possible there is no pre-image. The return value of `getPreimage` is about whether the operation is processed succesfully, not about whether a pre-image exists.